jobs/integration/validation.py

```diff
@@ -2205,32 +2206,82 @@ async def cleanup():
     await cleanup()
 
 
+@dataclass
+class NagiosAlerts:
+    alerts: Mapping[str, bs_ResultSet]
+    matchers: Mapping[str, Callable[[dict], bool]]
+
+    def __bool__(self):
+        """Successful if there are some alerts and all the matchers are True."""
+        if not self.alerts:
+            return False
+        return all(
+            self.matchers[severity](alerts) for severity, alerts in self.alerts.items()
+        )
+
+    def __getitem__(self, item: str) -> bs_ResultSet:
+        return self.alerts[item]
+
+    def __str__(self) -> str:
+        return ", ".join(
+            f"{severity}/{link.string}"
+            for severity, alerts in self.alerts.items()
+            for alert in alerts
+            for link in alert.find_all("a", recursive=False)
+        )
+
+
 @dataclass
 class NagiosApi:
     open: Callable
     url: str
 
-    def find_alerts(self, *severities):
+    async def find_alerts(self, **severities) -> NagiosAlerts:
         classes = {
             "critical": "statusBGCRITICAL",
             "pending": "statusPENDING",
             "ok": "statusOK",
         }
         url_data = self.open(self.url)
         soup = bs(url_data.read(), "html.parser")
-        return {
+        alerts = {
             severity: soup.find_all("td", class_=classes[severity.lower()])
             for severity in severities
         }
+        return NagiosAlerts(alerts, severities)
```

Comment:
this becomes an async method so it can be used by `retry_async_with_timeout`  It now also returns a `NagiosAlerts` object which can be evaluated for truthiness.